### PR TITLE
CDAP-4254 Added retry if the 'current' endpoint is not discoverable.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA: https://issues.cask.co/browse/CDAP-4254

When workflow is started, it takes some time to make the workflow http service discoverable even after it is marked as running. The workflow http service is responsible for serving the 'current' endpoint. When service is not discoverable it throws NotFoundException. Added retry logic in the test to handle the case when the 'current' endpoint is not discoverable.

Build is running here - http://builds.cask.co/browse/CDAP-DUT3235-1

### CODE REVIEW

File: cdap-client-tests/src/test/java/co/cask/cdap/client/ProgramClientTestRun.java [hunk 1]
```diff
@@ -212,7 +213,12 @@ private void testWorkflowCommand(final Id.Program workflow) throws Exception {
...
       public Integer call() throws Exception {
-        return programClient.getWorkflowCurrent(workflow.getApplication(), workflow.getId(), pid).size();
+        try {
+          return programClient.getWorkflowCurrent(workflow.getApplication(), workflow.getId(), pid).size();
+        } catch (NotFoundException e) {
+          // try again if the 'current' endpoint is not discoverable yet
+          return 0;
+        }
       }
     }, 20, TimeUnit.SECONDS, 100, TimeUnit.MILLISECONDS);
```
Comment (bdmogal) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/ProgramClientTestRun.java]: lets remove the last 2 params of this method so it uses the default 50 ms?

**Comment by bdmogal** [comment]
1 comment, rest LGTM

File: cdap-client-tests/src/test/java/co/cask/cdap/client/ProgramClientTestRun.java (hunk 1 above)
Comment (sagarkapare) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/ProgramClientTestRun.java]: Fixed.

**Comment by sagarkapare** [comment]
Thanks @bdmogal. Started new build here - http://builds.cask.co/browse/CDAP-DUT3235-2

**Comment by sagarkapare** [comment]
Build passed. Merging.

**Pull request merged**